lib/matplotlib/axes/_subplots.py

```diff
@@ -173,7 +173,9 @@ def _make_twin_axes(self, *kl, **kwargs):
         """
         from matplotlib.projections import process_projection_requirements
         if 'sharex' in kwargs and 'sharey' in kwargs:
-            raise ValueError("Twinned Axes may share only one axis.")
+            if (not (kwargs['sharex'] == self ) and
+                        not (kwargs['sharex'] == self)):
+                raise ValueError("Twinned Axes may share only one axis.")
```

Comment:
Are those line a trial for a 2-in-1 PR done on purpose regarding to #10585 ?

Reply:
Ooops, this shouldn't have been in this PR....